apps/wallet-mobile/src/features/Receive/useCases/RequestSpecificAmountScreen.tsx

```diff
@@ -48,7 +48,8 @@ export const RequestSpecificAmountScreen = () => {
 
   const handleOnChangeAmount = (amount: string) => {
     const edited = editedFormatter(amount)
-    if (Number(edited.replace(',', '.')) <= Number.MAX_SAFE_INTEGER) {
+    const numberOfDecimals = (edited.split('.')[1] ?? []).length
+    if (Number(edited) <= Number.MAX_SAFE_INTEGER && numberOfDecimals <= 6) {
```

Comment:
with portfolio this will get better, we will have formatters, parsers, static data without decisions on UI 🤞🏻 
```suggestion
    if (Number(edited) <= Number.MAX_SAFE_INTEGER && numberOfDecimals <= primaryTokenInfo.decimals ?? 0) {
```

Reply:
Yeah, I assumed so. Just preventing the UI issues with unlimited decimals.